scipy/signal/ltisys.py

```diff
@@ -678,119 +704,355 @@ def num(self):
+    state-space matrices) is very inefficient and may lead to numerical
+    inaccuracies.
+
+    If (numerator, denominator) is passed in for ``*system``, coefficients
+    for both the numerator and denominator should be specified in descending
+    exponent order (e.g., ``z^2 + 3z + 5`` would be represented as
+    ``[1, 3, 5]``).
+
+    Examples
+    --------
+    Construct the transfer function with a sampling time of 0.5 seconds:
 
+    .. math:: H(z) = \frac{z^2 + 3z + 3}{z^2 + 2z + 1}
 
-class ZerosPolesGain(lti):
+    >>> from scipy import signal
+    >>> num = [1, 3, 3]
+    >>> den = [1, 2, 1]
+    >>> signal.TransferFunction(num, den, 0.5)
+    TransferFunctionDiscrete(
+    array([ 1.,  3.,  3.]),
+    array([ 1.,  2.,  1.]),
+    dt: 0.5
+    )
     """
+    pass
+
+
+class ZerosPolesGain(LinearTimeInvariant):
+    r"""
     Linear Time Invariant system class in zeros, poles, gain form.
 
-    Represents the system as the transfer function
+    Represents the system as the continuous- or discrete-time transfer function
     :math:`H(s)=k \prod_i (s - z[i]) / \prod_j (s - p[j])`, where :math:`k` is
     the `gain`, :math:`z` are the `zeros` and :math:`p` are the `poles`.
+    `ZerosPolesGain` systems inherit additional functionality from the `lti`,
+    respectively the `dlti` classes, depending on which system representation
+    is used.
 
     Parameters
     ----------
     *system : arguments
-        The `ZerosPolesGain` class can be instantiated with 1 or 3 arguments.
-        The following gives the number of input arguments and their
+        The `ZerosPolesGain` class can be instantiated with 1 or 3
+        arguments. The following gives the number of input arguments and their
         interpretation:
 
-            * 1: `lti` system: (`StateSpace`, `TransferFunction` or
+            * 1: `lti` or `dlti` system: (`StateSpace`, `TransferFunction` or
               `ZerosPolesGain`)
             * 3: array_like: (zeros, poles, gain)
+    dt: float, optional keyword argument
+        Sampling time [s] of the discrete-time systems. Defaults to `None`
+        (continuous-time). Must be specified as a keyword argument, for
+        example, dt=0.1.
+
 
     See Also
     --------
-    TransferFunction, StateSpace, lti
+    TransferFunction, StateSpace, lti, dlti
     zpk2ss, zpk2tf, zpk2sos
 
     Notes
     -----
     Changing the value of properties that are not part of the
     `ZerosPolesGain` system representation (such as the `A`, `B`, `C`, `D`
     state-space matrices) is very inefficient and may lead to numerical
-    inaccuracies.
+    inaccuracies.  It is better to convert to the specific system
+    representation first. For example, call `sys = sys.to_ss()` before
+    accessing/changing the A, B, C, D system matrices.
+
+    Examples
+    --------
+    >>> from scipy import signal
+
+    # Transfer function: H(s) = 5(s - 1)(s - 2) / (s - 3)(s - 4)
```

Comment:
This doesn't render correctly. There should always be a blank line between lines terminal in/output and text. And then the ``#`` isn't needed. See https://circle-artifacts.com/gh/scipy/scipy/99/artifacts/0/home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.signal.ZerosPolesGain.html#scipy.signal.ZerosPolesGain for rendered html on this PR.

This comment applies to many docstrings in this PR.